core/meta/src/TCling.cxx

```diff
@@ -5283,8 +5283,8 @@ void TCling::UpdateClassInfoWithDecl(const void* vTD)
       }
       clang::QualType type( td->getTypeForDecl(), 0 );
 
-      auto declName=ND->getNameAsString().c_str();
-      if (!TClass::HasNoInfoOrEmuOrFwdDeclaredDecl(declName)){
+      auto declName = ND->getNameAsString();
```

Comment:
I was naively assuming it returned a `const &`. If it already returns a 
string, of course you are right.

-- 
Ciao
Giulio

On 27 May 2015, at 13:05, davidlt wrote:

>> @@ -5283,8 +5283,8 @@ void TCling::UpdateClassInfoWithDecl(const 
>> void* vTD)
>>     }
>>     clang::QualType type( td->getTypeForDecl(), 0 );
>>
>> -      auto declName=ND->getNameAsString().c_str();
>> -      if (!TClass::HasNoInfoOrEmuOrFwdDeclaredDecl(declName)){
>> +      auto declName = ND->getNameAsString();
>
> A string is constructed either way, the patch only extends lifetime of 
> the object till the end of the function.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/root/pull/81/files#r31122433